sw/device/tests/alert_handler_lpg_clkoff_test.c

```diff
@@ -0,0 +1,619 @@
+  };
+
+  dif_alert_handler_class_config_t class_configs[] = {class_config,
+                                                      class_config};
+
+  dif_alert_handler_class_t classes[] = {kDifAlertHandlerClassA,
+                                         kDifAlertHandlerClassB};
+  dif_alert_handler_config_t config = {
+      .alerts = alerts,
+      .alert_classes = alert_classes,
+      .alerts_len = array_idx,
+      .local_alerts = loc_alerts,
+      .local_alert_classes = loc_alert_classes,
+      .local_alerts_len = ARRAYSIZE(loc_alerts),
+      .classes = classes,
+      .class_configs = class_configs,
+      .classes_len = ARRAYSIZE(class_configs),
+      .ping_timeout = ping_timeout,
+  };
+
+  alert_handler_testutils_configure_all(&alert_handler, config,
+                                        kDifToggleEnabled);
+  // Enables alert handler irq.
+  CHECK_DIF_OK(dif_alert_handler_irq_set_enabled(
+      &alert_handler, kDifAlertHandlerIrqClassa, kDifToggleEnabled));
+
+  CHECK_DIF_OK(dif_alert_handler_irq_set_enabled(
+      &alert_handler, kDifAlertHandlerIrqClassb, kDifToggleEnabled));
+}
+
+/**
+ *  A utility function to enable/disable hintable and gateable clocks
+ */
+void set_peripheral_clock(const test_t *peripheral,
+                          dif_toggle_t new_clk_state) {
+  dif_toggle_t clk_state;
+  // LOG_INFO("Periphereal name = %s, is_hintable = %d,
+  // clk_index=%d",peripheral->name, peripheral->is_hintable,
+  // peripheral->clk_index);
+  if (peripheral->is_hintable) {
+    // Set the new clock state
+    CHECK_DIF_OK(dif_clkmgr_hintable_clock_set_hint(
+        &clkmgr, peripheral->clk_index, new_clk_state));
+    // Verify the new clock state
+    CHECK_DIF_OK(dif_clkmgr_hintable_clock_get_hint(
+        &clkmgr, peripheral->clk_index, &clk_state));
+    CHECK(clk_state == new_clk_state,
+          "intended_clk_state = %d, received_clk_state = %d", new_clk_state,
+          clk_state);
+
+  } else {
+    CHECK_DIF_OK(dif_clkmgr_gateable_clock_set_enabled(
+        &clkmgr, peripheral->clk_index, new_clk_state));
+    // Verify the new clock state
+    CHECK_DIF_OK(dif_clkmgr_gateable_clock_get_enabled(
+        &clkmgr, peripheral->clk_index, &clk_state));
+    CHECK(clk_state == new_clk_state,
+          "intended_clk_state = %d, received_clk_state = %d", new_clk_state,
+          clk_state);
+  }
+};
+
+/**
+ * A utility function to wait enough until the alert handler pings a peripheral
+ * alert
+ */
+void wait_enough_for_alert_ping() {
+  // wait enough
+  if (kDeviceType == kDeviceFpgaCw310) {
+    // NUM_ALERTS*2*margin_of_safety*(2**DW)*(1/kClockFreqPeripheralHz)
+    // (2**6)*2*4*(2**16)*(400ns) = 12.8s
+    busy_spin_micros(1000 * 12800);
+  } else if (kDeviceType == kDeviceSimDV) {
+    // NUM_ALERTS*2*margin_of_safety*(2**DW)*(1/kClockFreqPeripheralHz)
+    // (2**6)*2*4*(2**3)*(40ns) = 160us
+    busy_spin_micros(160);
+  } else {
+    LOG_FATAL("SUPPORTED PLATFORMS: DV and FPGA");
```

Comment:
Ah, so there's actually work that would need to be done to support Verilator here? In that case, I think it's definitely fine to not support Verilator, but the comment in the BUILD file should probably say something about it being more than just a timeout issue (otherwise someone might come along and just add a Verilator target, assuming the test should work).

Reply:
Thanks, Jade. I added more info into the BUILD file as well as into the `.c` code.